discovery/kubernetes/kubernetes_test.go

```diff
@@ -105,6 +105,7 @@ type k8sDiscoveryTest struct {
 
 func (d k8sDiscoveryTest) Run(t *testing.T) {
 	t.Helper()
+	t.Parallel()
```

Comment:
I think it's better to have this explicitly set for each test/subtest that needs it.

Reply:
All the tests under `discovery/kubernetes` populate the `k8sDiscoveryTest` struct and call `.Run(t)` on the object. 
I figured it's safe to put `t.Parallel()` in the helper as all the tests are normalized to use it. 

What downsides do you see by putting it here?

Reply:
It's yet another flag, I still think explicitly setting `t.Parallel()` is the safer/clearer way, it's verbose, but there will be less hidden magic.

Reply:
Makes sense @machine424 
I've removed the flag and `t.Parallel()` from the helper func, and added it to each test instead. 
Please re-review this PR.